dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -2201,22 +2341,42 @@ def dpnp_svd(
             s = dpnp.abs(s)
             return dpnp.sort(s)[..., ::-1]
 
-    uv_type = _common_type(a)
+    uv_type = (
```

Comment:
why do we need to calculate dtype based on `related_arrays`? what is the use case?

Reply:
Doesn't that mean we will use `_gesvd` with wider dtype than required, which may have negative performance drawback?